lib/Twig/Template.php

```diff
@@ -523,18 +523,21 @@ protected function getAttribute($object, $item, array $arguments = array(), $typ
         }
 
         // object property
-        if (self::METHOD_CALL !== $type && !$object instanceof self) { // Twig_Template does not have public properties, and we don't want to allow access to internal ones
-            if (isset($object->$item) || array_key_exists((string) $item, $object)) {
-                if ($isDefinedTest) {
-                    return true;
-                }
+        try {
+            if (self::METHOD_CALL !== $type && !$object instanceof self) { // Twig_Template does not have public properties, and we don't want to allow access to internal ones
+                if (isset($object->$item) || array_key_exists((string)$item, $object)) {
+                    if ($isDefinedTest) {
+                        return true;
+                    }
 
-                if ($this->env->hasExtension('sandbox')) {
-                    $this->env->getExtension('sandbox')->checkPropertyAllowed($object, $item);
-                }
+                    if ($this->env->hasExtension('sandbox')) {
+                        $this->env->getExtension('sandbox')->checkPropertyAllowed($object, $item);
+                    }
 
-                return $object->$item;
+                    return $object->$item;
+                }
             }
+        } catch (Twig_Sandbox_SecurityError $propertySandboxException) {
```

Comment:
Is there any particular case why the try is surrounding such a big piece of code?